### PR TITLE
Fix CSS order when merging type change bundles

### PR DESCRIPTION
**Pull request by devongovett**
This fixes a bug when merging type change bundles that could result in CSS being bundled in the wrong order. We need to merge the contents of later bundles into earlier bundles, not the other way around.

I think this was potentially a workaround for an issue with CSS modules where there was no dependency in the CSS asset to `composes` deps, only in the JS, which left the graph in a weird state. I don't remember why that was the case, but now the dependency is in both places which fixes the problem. The test output was also slightly incorrect before anyway.

### CODE REVIEW

File: packages/core/integration-tests/test/css.js [hunk 1]
```diff
@@ -22,17 +22,21 @@ describe('css', () => {
...
       },
     ]);
 
     let output = await run(b);
     assert.equal(typeof output, 'function');
     assert.equal(output(), 3);
+
+    let css = await outputFS.readFile(path.join(distDir, 'index.css'), 'utf8');
+    assert.ok(css.indexOf('.c {') < css.indexOf('.local {'));
+    assert.ok(css.indexOf('.local {') < css.indexOf('.index {'));
```
Comment (devongovett) [inline packages/core/integration-tests/test/css.js:39]: previous order was c, index, local. now order is c, local, index.

File: packages/core/integration-tests/test/css-modules.js [hunk 2]
```diff
@@ -353,6 +353,7 @@ describe('css modules', () => {
     assert(css.includes('height: 100px;'));
     assert(css.includes('height: 300px;'));
     assert(css.indexOf('_test') < css.indexOf('_intermediate'));
+    assert(css.indexOf('_intermediate') < css.indexOf('_composes5'));
```
Comment (devongovett) [inline packages/core/integration-tests/test/css-modules.js:356]: previously composes5 was first, now it's last as expected.

**Comment by parcel-benchmark** [comment]
## Benchmark Results
<details><summary>Kitchen Sink ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.56s | +23.00ms |
| Cached | 341.00ms | -45.00ms 🚀 |

#### Cold Bundles

| Bundle | Size | Difference | Time | Difference |
| --- | --- | --- | --- | --- |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 108.00ms | -146.00ms 🚀 |
| dist/legacy/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 109.00ms | -146.00ms 🚀 |
| dist/modern/parcel.7cdb0fad.webp | 102.94kb | +0.00b | 109.00ms | -147.00ms 🚀 |
| dist/legacy/index.2c76ad23.js | 1.66kb | +0.00b | 455.00ms | +29.00ms ⚠️ |
| dist/legacy/index.8aaa89c9.js | 1.20kb | +0.00b | 455.00ms | +28.00ms ⚠️ |
| dist/modern/index.6be20f01.js | 1.13kb | +0.00b | 455.00ms | +28.00ms ⚠️ |
| dist/legacy/index.html | 826.00b | +0.00b | 561.00ms | +35.00ms ⚠️ |
| dist/modern/index.html | 749.00b | +0.00b | 561.00ms | +41.00ms ⚠️ |
| dist/legacy/index.b8ae99ba.css | 94.00b | +0.00b | 283.00ms | +22.00ms ⚠️ |
| dist/modern/index.31cedca9.css | 94.00b | +0.00b | 283.00ms | +22.00ms ⚠️ |

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>React HackerNews ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 10.22s | +167.00ms |
| Cached | 467.00ms | +5.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>AtlasKit Editor ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 1.75m | -2.40s |
| Cached | 2.37s | -25.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

<details><summary>Three.js ✅</summary><p>

#### Timings

| Description | Time | Difference |
| --- | --- | --- |
| Cold | 7.41s | +109.00ms |
| Cached | 291.00ms | +2.00ms |

#### Cold Bundles

*No bundle changes detected.*

#### Cached Bundles

*No bundle changes detected.*

</p></details>

**[Click here to view a detailed benchmark overview.](https://benchmark.parceljs.org/benchmark/354129433631654487)**

**Review by AGawrys — APPROVED** [review]
All good, just realized we need to alter bundle size here as well as adding the asset, I'll add that in another PR

**Pull request merged**